src/DetectorConstruction.h

```diff
@@ -21,6 +21,12 @@ class DetectorConstruction: public G4VUserDetectorConstruction
     virtual ~DetectorConstruction();
 
   private:
+    G4LogicalVolume* detector_logic_vol;
```

Comment:
I like that you've made the detector volume a data member.  That will make your ConstructSDandField() function nice, since you'll have the pointer you need already saved.